### DISCUSSION TITLE
Apache.Arrow C# usage?

### DISCUSSION THREAD

**Discussion by eerhardt** [trmcnealy/VegaLite.NET#5] [General]
@trmcnealy - I was looking through the users of the [Apache.Arrow](https://www.nuget.org/packages/Apache.Arrow/) package, and this project came up.

Looking through the history, it looks like you were using the C# library, but have switched to the JavaScript library.

Can you tell me more about this? Specifically, was there something about the C# library that didn't meet your needs? If so, I'd like to hear about it. Maybe opening a [JIRA issue](https://issues.apache.org/jira/projects/ARROW/issues/ARROW-10844?filter=allopenissues) if possible, or just describe it to me, and I can log any issues.

Anyway, this project looks cool. It's great to see .NET devs making these projects for other .NET devs.

**Comment by trmcnealy** [comment]
I started with the javascript library and then later added Apache.Arrow because the data set was too large to plot, 30+ MB.  And you can't write that much data directly into the browser and have it functional.  Which is one of the downfalls of the Vega libraries, all the data inputs are json format. The Apache.Arrow library I'm sure is good on other languages but not for C#, it is not a direct port and therefore it is missing some of the functionality that the other languages have (Cuda IPC).  With that being said the only advantage I see of using Apache.Arrow would be if your going to host a single file with all of your data, in other words storing post analysis data online to share. For my projects, using it for interoperability really isn't that efficient, which is why I changed graph libraries and I can just keep everything in memory,  because I generate more data then I have stored as an initial input.
